201-site-to-site-vpn-fqdn-bgp/azuredeploy.json

```diff
@@ -0,0 +1,1109 @@
+      }
+    },
+    "localGatewayName22": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('prefix'),'localGateway22')]",
+      "metadata": {
+        "description": "Arbitrary name for gateway resource representing VPN gateway2-publicIP2"
+      }
+    },
+    "connectionName11-21": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('gateway1Name'),'-to-', parameters('gateway2PublicIP1Name'))]",
+      "metadata": {
+        "description": "Arbitrary name for the new connection between VPN gateway1 and the remote VPN Gateway2-public IP1"
+      }
+    },
+    "connectionName12-22": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('gateway1Name'),'-to-', parameters('gateway2PublicIP2Name'))]",
+      "metadata": {
+        "description": "Arbitrary name for the new connection between VPN gateway1 and the remote VPN Gateway2-public IP2"
+      }
+    },
+    "connectionName21-11": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('gateway2Name'),'-to-', parameters('gateway1PublicIP1Name'))]",
+      "metadata": {
+        "description": "Arbitrary name for the new connection between VPN gateway2 and the remote VPN Gateway1-public IP1"
+      }
+    },
+    "connectionName22-12": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('gateway2Name'),'-to-', parameters('gateway1PublicIP2Name'))]",
+      "metadata": {
+        "description": "Arbitrary name for the new connection between VPN gateway2 and the remote VPN Gateway1-public IP2"
+      }
+    },
+    "sharedKey": {
+      "type": "string",
+      "defaultValue": "_your_Pre-Shared-Secret",
+      "metadata": {
+        "description": "Shared key (PSK) for IPSec tunnels"
+      }
+    },
+    "vm1Name": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('prefix'),'vm1')]",
+      "metadata": {
+        "description": "name of the VM in subnet1 in VNet1"
+      }
+    },
+    "vm2Name": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('prefix'),'vm2')]",
+      "metadata": {
+        "description": "name of the VM in subnet1 in VNet2"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_B1s",
+      "allowedValues": [
+        "Standard_B1ls",
+        "Standard_B1s",
+        "Standard_DS1_v2",
+        "Standard_DS5_v2"
+      ],
+      "metadata": {
+        "description": "Size of the Virtual Machine"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "administrator username of the VMs"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "administrator password of the VMs"
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Premium_LRS"
```

Comment:
if you're going to allow for premium disk then the VM sizes must match... and alternative is to use StandardSSD which is supported by all VM sizes